#! /bin/sh /usr/share/dpatch/dpatch-run
## Makefile.dpatch by Jose Carlos Garcia Sogo <jsogo@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Patch Makefile to build under Debian GNU/Linux

@DPATCH@
diff -urNad asterisk-1.0.5/Makefile /tmp/dpep.jxj8cL/asterisk-1.0.5/Makefile
--- asterisk-1.0.5/Makefile	2005-01-28 21:15:18.000000000 +0100
+++ /tmp/dpep.jxj8cL/asterisk-1.0.5/Makefile	2005-01-28 23:28:29.000000000 +0100
@@ -18,42 +18,6 @@
 
 OSARCH=$(shell uname -s)
 
-ifeq (${OSARCH},Linux)
-PROC=$(shell uname -m)
-ifeq ($(PROC),x86_64)
-# You must have GCC 3.4 to use k8, otherwise use athlon
-PROC=k8
-#PROC=athlon
-OPTIONS+=-m64
-endif
-ifeq ($(PROC),sparc64)
-#The problem with sparc is the best stuff is in newer versions of gcc (post 3.0) only.
-#This works for even old (2.96) versions of gcc and provides a small boost either way.
-#A ultrasparc cpu is really v9 but the stock debian stable 3.0 gcc doesn't support it.
-#So we go lowest common available by gcc and go a step down, still a step up from
-#the default as we now have a better instruction set to work with. - Belgarath
-PROC=ultrasparc
-OPTIONS+=$(shell if $(CC) -mtune=$(PROC) -S -o /dev/null -xc /dev/null >/dev/null 2>&1; then echo "-mtune=$(PROC)"; fi)
-OPTIONS+=$(shell if $(CC) -mcpu=v8 -S -o /dev/null -xc /dev/null >/dev/null 2>&1; then echo "-mcpu=v8"; fi)
-OPTIONS+=-fomit-frame-pointer
-endif
-
-MPG123TARG=linux
-endif
-
-ifeq ($(findstring BSD,${OSARCH}),BSD)
-PROC=$(shell uname -m)
-endif
-
-# Pentium Pro Optimize
-#PROC=i686
-
-# Pentium & VIA processors optimize
-#PROC=i586
-
-#PROC=k6
-#PROC=ppc
-
 PWD=$(shell pwd)
 
 ######### More GSM codec optimization
@@ -63,7 +27,7 @@
 #K6OPT  = -DK6OPT
 
 #Tell gcc to optimize the asterisk's code
-OPTIMIZE+=-O6
+OPTIMIZE+=-O2
 
 #Include debug symbols in the executables (-g) and profiling info (-pg)
 DEBUG=-g #-pg
@@ -134,7 +99,7 @@
 CFLAGS+=$(shell if $(CC) -march=$(PROC) -S -o /dev/null -xc /dev/null >/dev/null 2>&1; then echo "-march=$(PROC)"; fi)
 endif
 
-CFLAGS+=$(shell if uname -m | grep -q ppc; then echo "-fsigned-char"; fi)
+CFLAGS+=$(shell if uname -m | grep -q "ppc\|arm\|s390\|s390x"; then echo "-fsigned-char"; fi)
 CFLAGS+=$(shell if [ -f /usr/include/osp/osp.h ]; then echo "-DOSP_SUPPORT -I/usr/include/osp" ; fi)
 
 ifeq (${OSARCH},FreeBSD)
@@ -163,7 +128,7 @@
 CFLAGS+=$(shell if [ -f /usr/local/include/zaptel.h ]; then echo "-DZAPTEL_OPTIMIZATIONS"; fi)
 CFLAGS+=$(shell if [ -f ../include/zaptel.h ]; then echo "-DZAPTEL_OPTIMIZATIONS"; fi)
 
-LIBEDIT=editline/libedit.a
+LIBEDIT=-Leditline -ledit
 
 ASTERISKVERSION=$(shell if [ -f .version ]; then cat .version; else if [ -d CVS ]; then if [ -f CVS/Tag ] ; then echo "CVS-`sed 's/^T//g' CVS/Tag`-`date +"%D-%T"`"; else echo "CVS-HEAD-`date +"%D-%T"`"; fi; fi; fi)
 HTTPDIR=$(shell if [ -d /var/www ]; then echo "/var/www"; else echo "/home/httpd"; fi)
@@ -533,7 +500,7 @@
 	fi 
 
 dont-optimize:
-	$(MAKE) OPTIMIZE= K6OPT= install
+	$(MAKE) OPTFLAGS= install
 
 valgrind: dont-optimize
 
